subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultResolvedArtifact.java

```diff
@@ -122,29 +122,33 @@ public boolean isResolveSynchronously() {
             // Don't bother resolving local components asynchronously
             return true;
         }
-        synchronized (this) {
-            return file != null || failure != null;
-        }
+        return file != null || failure != null;
     }
 
```

Comment:
Maybe add a method-level comment that his algorithm tries to minimize volatile reads and should not be simplified by inlining the local variables.

Reply:
Indeed.